server/uptime-calculator.js

```diff
@@ -307,12 +437,41 @@ class UptimeCalculator {
                 up: 0,
                 down: 0,
                 avgPing: 0,
+                minPing: 0,
+                maxPing: 0,
             });
         }
 
         return dailyKey;
     }
 
+    /**
+     * Convert timestamp to key
+     * @param {dayjs.Dayjs} datetime Datetime
+     * @param {"day" | "hour" | "minute"} type the type of data which is expected to be returned
+     * @returns {number} Timestamp
+     * @throws {Error} If the type is invalid
+     */
+    getKey(datetime, type) {
+        let key;
+        // Get the stats key based on the type
+        switch (type) {
+            case "day":
+                key = this.getDailyKey(datetime);
+                break;
+            case "hour":
+                key = this.getHourlyKey(datetime);
+                break;
+            case "minute":
+                key = this.getMinutelyKey(datetime);
+                break;
+            default:
+                throw new Error("Invalid type");
+        }
+
+        return key;
```

Comment:
I think this can be simplified

```suggestion
        switch (type) {
            case "day":
                return this.getDailyKey(datetime);
            case "hour":
                return this.getHourlyKey(datetime);
            case "minute":
                return this.getMinutelyKey(datetime);
            default:
                throw new Error("Invalid type");
        }
```